### DISCUSSION TITLE
Export drawings to powerpoint as objects

### DISCUSSION THREAD

**Discussion by vhristev** [excalidraw/excalidraw#8090] [Q&A]
Hey team, 

Do you have any idea how I can export drawings from Excalidraw to PowerPoint as objects, vectors, etc.?

I'm totally in love with Excalidraw. For the last three years, I've made all my presentations with it, but just the drawings. I'm using PowerPoint "transform" for slides to make drawings move/resize...etc. 

The main problem is that I draw in Excali, copy it as an image, and paste it into PowerPoint. Because it's an image, not an object, it's tough to make that presentation. It's super time-consuming. Is there a way or feature request for that? 

You may suggest Excalidraw presentations, but I cannot make those animations the way I want them to be. Back in the days in VMware, we had icons packed, which are objects. 

Here is a video to see the animation: https://youtube.com/watch?v=UNlxLgsNd2U&t=365 

Here are some lectures:
https://github.com/DojoBits/cncf/blob/main/cncf-bulgaria-meetup-2024-may.pdf
https://github.com/vhristev/cncf-bulgaria-meetup-2024-feb/blob/main/CNCF-Bulgaria-Meetup-cotainers-security-final.pdf
https://github.com/vhristev/OpenFest2022/blob/main/000_Full_presentation.pdf

**Comment by whitehawkJulie** [comment]
Does this help? https://github.com/excalidraw/excalidraw/issues/3947

**Comment by vhristev** [comment]
@whitehawkJulie, sorry for the delay in getting the notification. 

So, I checked #3947, and I need something else. The Export to PPTX feature in Excalidraw+ just exports slides as pictures and generate PPTX file. 

My goal is to export a **vector object** from Excalidraw to PPTX. I want to be able to modify the object inside PPTX because the animations I'm using are from PPTX. 

As a workaround, I export every drawing as a PNG and use it in PPTX, but it's very time-consuming to make the presentations because I need to align objects that I picture, not o can modify them. 

@maielo, is there any plan to export objects from excalidraw to PPTX

**Reply by maielo** [reply]
@vhristev this sounds interesting! Our initial idea was to have the additional option to present Excalidraw slides or export them as is. 

We need to first figure out how to add them as objects to pptx, going to create an issue for it.

**Reply by Oddly** [reply]
This would be great, would love to export Excalidraw slides as pptx and then add the capabilities of Powerpoint (transitions/animations) to it.

**Reply by Moulick** [reply]
Definitely needed. I just made a PPT but had diagram in excalidraw. Took me lot of time to export groups of objects and align all of them

**Reply by D4D-Developer** [reply]
hey @vhristev, I just tried it with pptgenjs lib and it worked.
You need to have mapping of excalidraw widgets types with pptgenjs widgets and try.

imp: I did everything by asking co-pilot and happily it did for me. 

Considering frames as Slides and export.

**Reply by vhristev** [reply]
Hey @D4D-Developer , 

Not sure what `pptgenjs` is. If we  [talk about that](https://gitbrent.github.io/PptxGenJS/), I'm not sure what the step-by-step process is. If you could be more specific, it would be great. 

Currently, what I'm doing is:
- From Excalidraw export selected or everything:
  - Without background
  - Scene embedded
  - SVG 
  
<img width="1489" height="621" alt="SCR-20251027-ircr-2" src="https://github.com/user-attachments/assets/04c9971b-7daf-404e-99ec-79eb37e06cc8" />


In PowerPoint I just drag and drop the SVG and then select it and "Ungroup" 

<img width="1032" height="1069" alt="Screenshot 2025-10-27 at 9 21 36" src="https://github.com/user-attachments/assets/4450811f-6f00-4ddc-b809-b4dc732ea231" />

This converts the image to objects. Just to say it's not a perfect object, but it's much, much better than dealing with images.


<img width="1047" height="1107" alt="Screenshot 2025-10-27 at 9 21 54" src="https://github.com/user-attachments/assets/0f730c29-9ff6-4f08-9264-57a377066c79" />

**Reply by D4D-Developer** [reply]
@vhristev - apologies for the late response.
If you are using cursor then it becomes bit easy to implement. How? Read below-

If you observe, widget "types" of excalidraw widgets and microsoft powerpoint widgets are different. You need to first have correct mapping of these widget "types" and then export the widgets to ppt using pptgenjs(_easily doable by cursor, tried and tested_).

**Imagine**, you have Frames(_a widget in excalidraw_) in excalidraw canvas which you consider as Slides(_Should have fixed size, cursor can help here_) and then think of the widgets inside the Frame as Slide widgets and then export.

Steps-
1- Ask cursor to plot frames in canvas having Slide's size.
2- Now, ask Cursor to go through codebase, documentation of pptgenjs as what are the widget "types".
3- Ask it to provide mapping/converter of widgets from excalidraw to pptgenjs then only it can export as pptx.
4- Once you have the mapping, you can ask Cursor to export your canvas frames as Slides in pptx. 
Done.

P.S.-
You might observe some widgets are exporting wrong or they are not as per the canvas. For ex- "Line" widget exports as "Line" widget but it should be as "arc" widget. So, You can ask cursor to correct all of those issues 1 by 1. **Thats it**. 

Hope you are able to implement it like above. I have already done it using Cursor.

Feel free to ask further.

**Comment by vhristev** [comment]
Anyone ?

@maielo any change that idea is in the project roadmap plan ?

**Comment by vhristev** [comment]
Anyone.

**Reply by maielo** [reply]
https://github.com/excalidraw/excalidraw/discussions/8090#discussioncomment-10178633

**Reply by hxh103** [reply]
have you tried importing as svg an importing it to Powerpoint? or converting svg to EMF and then importing? Also exporting single item at a time seems to work better as ungrouping the svg may make your image not appear very nice depending on several factors. For example, text does not do well once ungrouped.

**Comment by DoctorMG** [comment]
Thanks for your proposals - but I think it is obvious, that this is a mess (sorry - don't want to blame helpers - I mean the technical situation). Clicks and clicks and clicks and than an export, navigating through the file system with the export, navigating again from powerpoints import dialog box for the import - (and I am sure one of both tools will always forget the last used folder) - select the file type, select then the files. And this feels even more annoying than my long sentence read.

What is requried is: select + ctrc c + ctrl v. Then there would be a workflow with really valuable results. The compromises are fine for one slide, maybe two. But honestly it is killing every serious work approach when this sideload makes more mouse miles than the real work, isn't it? 

In a nutshell: I would vote for this request.

**Comment by Momoru2002** [comment]
Hey team,

There's a way to export drawings from Excalidraw to PowerPoint as editable objects/vectors, not just a flat image.

Here's how: in Excalidraw, right-click your drawing and choose "Copy to clipboard as SVG" (or export as SVG from the export menu), instead of PNG. The difference is that PNG is just a grid of pixels, while SVG stores the actual vector shape data, so it can be "broken apart" into real objects.

After that, paste/insert the SVG into PowerPoint. Click on it, a "Graphics Format" tab will show up in the ribbon, then click "Convert to Shape". PowerPoint will ask for confirmation, just click Yes. After that, right-click again and choose "Ungroup" (sometimes you need to do it twice) - now the drawing is broken down into real PowerPoint shapes that you can edit, recolor, and most importantly, use with the Morph/Transform effect for animations.

Note that "Convert to Shape" is only available in PowerPoint 2019 / Office 365 and newer, older versions don't have it.

Also keep in mind the result isn't always perfect - for simple Excalidraw drawings (lines, arrows, boxes) it usually converts cleanly, but the more complex the drawing (lots of text, gradients, etc.), the more manual cleanup you might need after converting. Text inside the SVG sometimes gets converted into paths/shapes too, rather than staying as editable text.

But this is still way better than the usual "copy as image" approach, since that gives you a PNG (raster), and PowerPoint can't convert a raster image into shapes - the feature only works with vector graphics like SVG.

Hope this helps!

**Comment by DoctorMG** [comment]
@Momoru2002  
Thanks, I tried this right now, newest Excel-Version and cannot manage it. The difference PNG/SVG is clear. And the menu item in the context menu of Excalidraw is clearly named - until copying some elements in the clipboard it works. 

But at receiver software I cannot reproduce your solution. I get only a textbox with the SVG code inside and the activated ribbon is the one for shapes. I cannot find the "convert" command. I also inspected the insert options: sometimes here we can decide if something should come as graphic, text, image, OLE etc. - but in PPT Version 2608 / 365 I have only this text-in-textbox way. 

Since we are now with SVG I tried to go via Inkscape - there I could not insert anything :-( 
And before giving up I tried Adobe Illustrator: there I could paste in the SVG directly and editable. ... and from there I could copy the graphics again and this provided really excalidraw figures as editable graphics in PowerPoint.

From there on I see, what you mean with "isn't always" perfect. Yes, the handdrawn style is transfered by using paths. So a Excalidraw rectangle is not brought as a PowerPoint shape rectangle. Instead we receive a bunch of lines with points. That would not matter - but based on this it is not possible to change the fill color (since there is no fill area - just the path, pretending some rectangular thing but it is not). Lines/arrows don't snap etc. 

So the fantasy to use Excalidraw as a kind of fancyfying driver for some advanced PPT style does not come true. Or at least: it may not be impossible - but an Excalidraw look in PowerPoint has to be paid with a lot of restrictions. Unfortuately 😞
